liquigraph-core/src/main/java/org/liquigraph/core/io/LockGuardian.java

```diff
@@ -0,0 +1,92 @@
+package org.liquigraph.core.io;
+
+import org.liquigraph.core.exception.LiquigraphLockException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.UUID;
+
+public class LockGuardian {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LockGuardian.class);
+
+    /**
+     * Makes sure the guardian cleans up only its own Lock
+     */
+    private final String uuid;
+
+    public LockGuardian() {
+        uuid = UUID.randomUUID().toString();
+    }
+
+    /**
+     * This guardian tries to create a unique lock and
+     * fails fast if one or more is already present.
+     *
+     * Callers must not try to recover from the aforementioned errors.
+     *
+     * @param connection JDBC connection to the configured instance
+     * @throws LiquigraphLockException if the unicity constraint cannot be created or the lock creation fails
+     */
+    public void lock(Connection connection) {
+        registerLockRemovalHook(connection);
+        ensureLockUnicity(connection);
+        tryWriteLock(connection);
+    }
+
+    private void registerLockRemovalHook(final Connection connection) {
+        Runtime.getRuntime().addShutdownHook(new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try (PreparedStatement statement = connection.prepareStatement(
+                    "MATCH (lock:__LiquigraphLock {uuid:{1}}) DELETE lock")) {
+
+                    statement.setString(1, uuid);
+                    statement.execute();
```

Comment:
yep, noticed it too, just fixed by https://github.com/fbiville/liquigraph/commit/0f83feaecc741d931cb238d76428ecc0e4cd4ad0